src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+                    lastEle = itrN.previous();
+                if ( lastEle.getOperator() == CigarOperator.SOFT_CLIP &&
+                        lastEle.getLength() >= MIN_SOFT_CLIP_LEN &&
+                        highQualityRegion(read.getBaseQualities(), read.getLength()-lastEle.getLength()) )
+                    return new EventLocus(read.getEnd()-SOFT_CLIP_LOCUS_WIDTH/2, SOFT_CLIP_LOCUS_WIDTH);
+            }
+
+            int locus = read.getStart();
+            for ( CigarElement ele : cigarElements )
+            {
+                CigarOperator op = ele.getOperator();
+                if ( op == CigarOperator.DELETION || op == CigarOperator.INSERTION )
+                    if ( ele.getLength() >= MIN_INDEL_LEN )
+                        return new EventLocus(locus-SOFT_CLIP_LOCUS_WIDTH/2, SOFT_CLIP_LOCUS_WIDTH);
+                if ( op.consumesReferenceBases() )
+                    locus += ele.getLength();
+            }
+            return null;
+        }
+
+        private static boolean highQualityRegion( final byte[] quals, int idx )
+        {
+            for ( final int end = idx+MIN_SOFT_CLIP_LEN; idx != end; ++idx )
+                if ( quals[idx] < MIN_QUALITY )
+                    return false;
+
+            return true;
+        }
+    }
+
+    // Class to detect reads that are part of a discordant pair.
+    // Concordant pairs are mapped to opposite strands of a reference sequence at a distance that is consistant with the
+    // library fragment length.
+    // Discordant pairs aren't concordant, except that we're also ignoring reads with mates mapped to other contigs.
+    // Class assumes that all reads presented to it are mapped.
+    private static final class DiscordantPairDetector
+    {
+        private static final int FUNKY_PAIR_LOCUS_WIDTH = 100; // uncertainty in event locus for discordant pair
+        //TODO: Above should depend on the fragment size distribution
+
+        public EventLocus getLocusIfDiscordant( GATKRead read )
+        {
+            if ( read.mateIsUnmapped() )
+            {
+                int locus = read.isReverseStrand() ? read.getStart() - FUNKY_PAIR_LOCUS_WIDTH : read.getEnd();
+                return new EventLocus(locus,FUNKY_PAIR_LOCUS_WIDTH);
+            }
+            else if ( read.getContig() == read.getMateContig() )
+            {
+                if ( read.isReverseStrand() == read.mateIsReverseStrand() || !read.isProperlyPaired() )
+                {
+                    int locus = read.getFragmentLength() < 0 ? read.getStart() - FUNKY_PAIR_LOCUS_WIDTH : read.getEnd();
+                    return new EventLocus(locus,FUNKY_PAIR_LOCUS_WIDTH);
+                }
+            }
+
+            return null;
+        }
+    }
+
+    // A class that acts as a filter for a stream of reads.  It passes only those that are a part of a putative
+    // breakpoint cluster.
+    private static final class BreakpointClusterer implements Iterator<GATKRead>, Iterable<GATKRead>
```

Comment:
it's a bit deceptive that this is Iterable<> since it's not restartable.  Does it have to be iterable?  I would either, add a note, make it actually repeatably iterable, or stop implmenting `Iterable`